src/applications/proxy-rewrite/proxy-rewrite-entry.jsx

```diff
@@ -42,89 +50,126 @@ function createMutationObserverCallback() {
   return function mutationObserved(mutations) {
     mutations.forEach(mutation => {
       Array.from(mutation.addedNodes).forEach(node => {
+        const addedNode = node;
+
         if (node.nodeType === Node.ELEMENT_NODE) {
-          if (node.tagName === 'BODY') {
-            node.classList.add('merger');
-          } else if (node.matches(deprecatedSelector)) {
-            node.classList.add('brand-consolidation-deprecated');
+          if (addedNode.tagName === 'BODY') {
+            addedNode.classList.add('merger');
+            addedNode.style.fontSize = '12px';
+          } else if (addedNode.matches(deprecatedSelector)) {
+            addedNode.classList.add('brand-consolidation-deprecated');
           }
         }
       });
     });
   };
 }
 
-function activateHeaderFooter() {
-  // Set up elements for the new header and footer
-  const headerContainer = document.createElement('div');
+function renderHeader(megaMenuData, headerContainer = null) {
+  const container =
+    headerContainer || document.querySelector('.ts-header-container');
   const skipLink = document.getElementById('skiplink');
-  headerContainer.innerHTML = headerPartial;
-  headerContainer.classList.add('consolidated');
+  const isMobile = window.innerWidth < 768;
 
-  const footerContainer = document.createElement('div');
-  footerContainer.innerHTML = footerPartial;
-  footerContainer.classList.add('consolidated');
+  if (container) {
+    if (isMobile) {
+      container.innerHTML = getMobileHeaderHtml(megaMenuData);
+    } else {
+      container.innerHTML = getDesktopHeaderHtml(megaMenuData);
+    }
 
-  if (skipLink) {
-    skipLink.after(headerContainer);
-  } else if (document.body.firstChild) {
-    document.body.firstChild.before(headerContainer);
+    if (skipLink) {
+      skipLink.after(container);
+    } else if (document.body.firstChild) {
+      document.body.firstChild.before(container);
+    }
   }
-  document.body.appendChild(footerContainer);
+
+  addHeaderEventListeners();
+
+  const searchParentElement = isMobile
```

Comment:
This section allows us to inject the Search and SignInModal functionality as React apps (into our plain HTML Teamsites code).